MarvelApiClient/src/main/java/com/karumi/MarvelApiClient/CharacterApiClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *   Copyright (C) 2015 Karumi.
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package com.karumi.marvelapiclient;
+
+import com.karumi.marvelapiclient.model.CharacterDto;
+import com.karumi.marvelapiclient.model.CharactersDto;
+import com.karumi.marvelapiclient.model.CharactersQuery;
+import com.karumi.marvelapiclient.model.MarvelResponse;
+import java.util.Map;
+import retrofit.Call;
+
+/**
+ * This api give access to marvel Characters. You need a valid {@link MarvelApiClient}
+ */
+public class CharacterApiClient {
+  private final MarvelApiClient marvelApiClient;
+
+  public CharacterApiClient(MarvelApiClient marvelApiClient) {
+    this.marvelApiClient = marvelApiClient;
+  }
+
+  public MarvelResponse<CharactersDto> getAll(int offset, int limit) throws MarvelApiException {
+    CharacterApiRest api = marvelApiClient.getApi(CharacterApiRest.class);
+
+    CharactersQuery query =
+        CharactersQuery.Builder.create().withOffset(offset).withLimit(limit).build();
+    return getAll(query);
+  }
+
+  public MarvelResponse<CharactersDto> getAll(CharactersQuery charactersQuery)
+      throws MarvelApiException {
+    CharacterApiRest api = marvelApiClient.getApi(CharacterApiRest.class);
+
+    Map<String, Object> queryAsMap = charactersQuery.getAsMap();
+    Call<MarvelResponse<CharactersDto>> call = api.getCharacters(queryAsMap);
+    return marvelApiClient.execute(call);
+  }
+
+  public MarvelResponse<CharacterDto> getCharacter(String characterId) {
+    if (characterId == null || "".equals(characterId)) {
+      throw new IllegalArgumentException("The CharacterId must not be null or empty");
+    }
+    CharacterApiRest api = marvelApiClient.getApi(CharacterApiRest.class);
+
+    Call<MarvelResponse<CharactersDto>> call = api.getCharacter(characterId);
+    MarvelResponse<CharactersDto> characters = marvelApiClient.execute(call);
```

Comment:
I have the feeling this ``MarvelApiClient`` is doing too much, is kind of weird that: 

1. We are asking ``MarvelApiClient`` to get the rest api
2. We are asking the rest api for a call
3. We are asking ``MarvelApiClient`` again to execute the call I'm getting from the rest api

We should either merge these operations into a single call in ``MarvelApiClient`` or separate both concepts